psakefile.ps1

```diff
@@ -0,0 +1,104 @@
+  }
+  $syntaxPath = Join-Path $PSScriptRoot 'syntaxes/puppet.tmLanguage'
+  if (Test-Path -Path $syntaxPath) {
+    Remove-Item -Path $syntaxPath -Recurse -Force
+  }
+}
+
+task VendorEditorServices {
+  if ($config.editorComponents.editorServices.githubuser) {
+    $githubuser = $config.editorComponents.editorServices.githubuser
+  }
+  else {
+    $githubuser = 'lingua-pupuli'
+  }
+  if ($config.editorComponents.editorServices.githubrepo) {
+    $githubrepo = $config.editorComponents.editorServices.githubrepo
+  }
+  else {
+    $githubrepo = 'puppet-editor-services'
+  }
+
+  if ($config.editorComponents.editorServices.directory) {
+    throw "not implemented"
+  }
+
+  if ($config.editorComponents.editorServices.release) {
+    $releasenumber = $config.editorComponents.editorServices.release
+    $uri = "https://github.com/${githubuser}/${githubrepo}/releases/download/${releasenumber}/puppet_editor_services_${releasenumber}.zip";
+  }
+  elseif ($config.editorComponents.editorServices.githubref) {
+    $githubref = $config.editorComponents.editorServices.githubref;
+    $uri = "https://github.com/${githubuser}/${githubrepo}/archive/${githubref}.zip"
+  }
+  else {
+    throw "Unable to vendor Editor Serices. Missing a release, directory, or git reference configuration item"
+  }
+
+  Invoke-RestMethod -Uri $uri -OutFile $languageServerZip
+  Expand-Archive -Path $languageServerZip -DestinationPath $langaugeServerPath
+  Remove-Item -Path $languageServerZip -Force
+}
+
+task VendorEditorSyntax {
+  if ($config.editorComponents.editorSyntax.githubuser) {
+    $githubuser = $config.editorComponents.editorSyntax.githubuser
+  }
+  else {
+    $githubuser = 'lingua-pupuli'
+  }
+  if ($config.editorComponents.editorSyntax.githubrepo) {
+    $githubrepo = $config.editorComponents.editorSyntax.githubrepo
+  }
+  else {
+    $githubrepo = 'puppet-editor-syntax'
+  }
+
+  if ($config.editorComponents.editorSyntax.directory) {
+    $source = Join-Path ($config.editorComponents.editorSyntax.directory, 'syntaxes/puppet.tmLanguage')
+    Copy-Item -Path $source -Destination $syntaxFilePath
+    return
+  }
+
+  if ($config.editorComponents.editorSyntax.release) {
+    $releasenumber = $config.editorComponents.editorSyntax.release
+    $uri = "https://github.com/${githubuser}/${githubrepo}/releases/download/${releasenumber}/puppet.tmLanguage";
+  }
+  elseif ($config.editorComponents.editorSyntax.githubref) {
+    $githubref = $config.editorComponents.editorSyntax.githubref;
+    $uri = "https://raw.githubusercontent.com/${githubuser}/${githubrepo}/${githubref}/syntaxes/puppet.tmLanguage"
+  }
+  else {
+    throw "Unable to vendor Editor Syntax.  Missing a release, directory, or git reference configuration item"
+  }
+
+  Invoke-RestMethod -Uri $uri -OutFile $syntaxFilePath
+}
+
+task CompileTypeScript {
+  exec { tsc -p ./ }
+}
+
+task Bump {
+  exec { npm version --no-git-tag-version $packageVersion }
```

Comment:
Not sure we need this?

Reply:
We need this to increase the version for artifact storage